### PR TITLE
Task/enable autoformatting

### PR DESCRIPTION
**Pull request by Graham42**
Closes: https://github.com/englercj/tsd-jsdoc/issues/84

I've broken this down into separate commits so the configuration / code changes can be looked at separately from the formatting changes.

I tried to configure prettier to respect the existing code style. Let me know your thoughts or if there's any [other options](https://prettier.io/docs/en/options.html) you'd like me to set. If so I can rebase and re-do the formatting commit.

### CODE REVIEW

File: src/PropTree.ts [hunk 1]
```diff
@@ -1,28 +1,25 @@
 import { Dictionary } from './Dictionary';
 import { warn } from './logger';
 
-export interface IPropDesc
-{
+export interface IPropDesc {
```
Comment (englercj) [inline src/PropTree.ts]: I would like to keep the previous brace style, I don't like K&R style.

File: src/PropTree.ts [hunk 2]
```diff
@@ -36,40 +33,35 @@ export class PropTree
...
 
         // build the tree of props
-        for (let i = 0; i < props.length; ++i)
-        {
+        for (let i = 0; i < props.length; ++i) {
             const prop = props[i];
 
-            if (!prop || !prop.name)
-                continue;
+            if (!prop || !prop.name) continue;
```
Comment (englercj) [inline src/PropTree.ts]: I'd rather keep these as separate lines.

File: src/PropTree.ts [hunk 3]
```diff
@@ -36,40 +33,35 @@ export class PropTree
...
 
             const obj = this.nodes[prop.name];
 
-            if (!obj)
-            {
-                warn('Failed to find dot-notation property in map. This is likely a bug.');
+            if (!obj) {
+                warn(
+                    'Failed to find dot-notation property in map. This is likely a bug.',
+                );
```
Comment (englercj) [inline src/PropTree.ts]: This seems like a weird change.

File: src/create_helpers.ts [hunk 4]
```diff
@@ -1,67 +1,77 @@
...
+        for (let i = children.length - 1; i >= 0; --i) {
             const child = children[i];
-            if (!validate(child))
-            {
-                warn(`Encountered child that is not a ${msg}, this is likely due to invalid JSDoc.`, child);
+            if (!validate(child)) {
+                warn(
+                    `Encountered child that is not a ${msg}, this is likely due to invalid JSDoc.`,
+                    child,
+                );
```
Comment (englercj) [inline src/create_helpers.ts]: Maybe there is a max line length that is too short, or a minimum required params to make it do this? I don't mind splitting params to different lines, but this is happening too often.

File: src/create_helpers.ts [hunk 5]
```diff
@@ -1,67 +1,77 @@
...
-            {
-                warn('Encountered child that is not a supported declaration, this is likely due to invalid JSDoc.', child);
+            if (
+                !ts.isClassDeclaration(child) &&
+                !ts.isInterfaceDeclaration(child) &&
+                !ts.isFunctionDeclaration(child) &&
+                !ts.isEnumDeclaration(child) &&
+                !ts.isModuleDeclaration(child) &&
+                !ts.isTypeAliasDeclaration(child) &&
+                !ts.isVariableStatement(child)
```
Comment (englercj) [inline src/create_helpers.ts]: Definitely prefer my previous style of having the first item be on the same line as if, and have the operator precede the statement.

File: src/create_helpers.ts [hunk 6]
```diff
@@ -149,27 +169,30 @@ export function createFunction(doclet: IFunctionDoclet): ts.FunctionDeclaration
...
+    doclet: IFunctionDoclet,
+): ts.MethodDeclaration {
     const mods: ts.Modifier[] = [];
     const params = createFunctionParams(doclet);
     const type = createFunctionReturnType(doclet);
     const typeParams = resolveTypeParameters(doclet);
 
-    if (!doclet.memberof)
-        mods.push(declareModifier);
+    if (!doclet.memberof) mods.push(declareModifier);
```
Comment (englercj) [inline src/create_helpers.ts]: control flow statements should always be followed by a newline.

File: src/create_helpers.ts [hunk 7]
```diff
@@ -184,76 +207,79 @@ export function createClassMethod(doclet: IFunctionDoclet): ts.MethodDeclaration
...
+            undefined, // body
+        ),
+    );
 }
 
-export function createInterfaceMethod(doclet: IFunctionDoclet): ts.MethodSignature
-{
+export function createInterfaceMethod(
+    doclet: IFunctionDoclet,
+): ts.MethodSignature {
```
Comment (englercj) [inline src/create_helpers.ts]: Yeah this is happening too much, the previous format was fine.

**Comment by englercj** [comment]
A lot of changed here, gave it a quick skim and commented on some of the major changes I saw. I think the goal of this PR would be to get the auto formatter to be making as few changes to the current code as possible, which means it is matching the current style as much as possible.

File: src/PropTree.ts (hunk 1 above)
Comment (Graham42) [reply src/PropTree.ts]: This isn't an option with prettier.

File: src/PropTree.ts (hunk 2 above)
Comment (Graham42) [reply src/PropTree.ts]: In order to keep these as separate lines with prettier would need to add in braces. 

Prettier [wont do that out of principle](https://prettier.io/docs/en/rationale.html#what-prettier-is-not-concerned-about) but there's an [eslint rule](https://eslint.org/docs/rules/curly) that can auto-add this.

File: src/PropTree.ts (hunk 3 above)
Comment (Graham42) [reply src/PropTree.ts]: This is probably because of the line length that it gets wrapped.

File: src/create_helpers.ts (hunk 5 above)
Comment (Graham42) [reply src/create_helpers.ts]: This isn't an option with prettier

File: src/create_helpers.ts (hunk 4 above)
Comment (Graham42) [reply src/create_helpers.ts]: Depending the outcome of other comments, if this is the only this stopping us going forward, I can create a branch with a line length of 100 to compare.

File: src/create_helpers.ts (hunk 6 above)
Comment (Graham42) [reply src/create_helpers.ts]: Same as above, would need to add in braces.

File: src/create_helpers.ts (hunk 7 above)
Comment (Graham42) [reply src/create_helpers.ts]: Same as above, I can adjust the line length after resolving other comments.

**Comment by Graham42** [comment]
I've updated the prettier config to 100 printWidth and added eslint with an autofix rule to add braces for if statements.

Overall, there a couple changes that we can't configure. If those are a deal breaker then kinda out of luck. My personal opinion is that the format matters less than the ability to autoformat and not have to think about it in the future.

**Comment by englercj** [comment]
> Overall, there a couple changes that we can't configure. If those are a deal breaker then kinda out of luck. 

Yeah this is generally why I don't use auto-format utilities, they rarely can match the style I want.

> My personal opinion is that the format matters less than the ability to autoformat and not have to think about it in the future.

I can understand that point of view, but I don't share it.

Thanks for taking a stab at this, sorry it didn't work out. I'd prefer to keep the code as-is.

**Pull request closed without merging**